roxie/udplib/udpsha.cpp

```diff
@@ -34,18 +34,38 @@ using roxiemem::IDataBufferManager;
 
 IDataBufferManager *bufferManager;
 
+// All exported udp configuration options - these values provide the default values
+#ifdef TEST_DROPPED_PACKETS
+bool udpDropDataPackets = false;
+unsigned udpDropFlowPackets[flowType::max_flow_cmd] = {};
+unsigned flowPacketsSent[flowType::max_flow_cmd] = {};
+#endif
+
 bool udpTraceFlow = false;
 bool udpTraceTimeouts = false;
+
 unsigned udpTraceLevel = 0;
 unsigned udpFlowSocketsSize = 131072;
 unsigned udpLocalWriteSocketSize = 1024000;
 unsigned udpStatsReportInterval = 60000;
 
-#ifdef TEST_DROPPED_PACKETS
-bool udpDropDataPackets = false;
-unsigned udpDropFlowPackets[flowType::max_flow_cmd] = {};
-unsigned flowPacketsSent[flowType::max_flow_cmd] = {};
-#endif
+unsigned udpOutQsPriority = 0;
+unsigned udpMaxRetryTimedoutReqs = 0;   // 0 means off (keep retrying forever)
+
+//The following control the timeouts within the udp layer.  All timings are in milliseconds, but I suspect some of these should possibly be sub-millisecond
+unsigned udpFlowAckTimeout = 2;         // the maximum time that it is expected to take to receive an acknowledgement of a flow message (when one is sent) - should be small
+unsigned updDataSendTimeout = 20;       // how long to receive the maximum amount of data, ~100 packets of 8K should take 10ms on a 1Gb network. Timeout for assuming send_complete has been lost
+unsigned udpRequestTimeout = 20;        // How long from a request for a permit until the permit is granted - timeout for assuming a ok_to_send has been lost.
+unsigned udpPermitTimeout = 50;         // How long is a grant expected to last before it is assumed lost?
+unsigned udpResendTimeout = 0;          // How long should elapse after a packet has been sent before we assume it is lost.
+                                        // 0 means they are unlikely to be lost, so worth resending as soon as it appears to be missing - trading duplicate packets for delays
+
+unsigned udpMaxPendingPermits = 1;      // default to 5?
+bool udpResendAllMissingPackets = false;  // Should probably default to true
```

Comment:
Dubious about what this means, whether it defeats the overlapping of multiple bursts from a single sender, and what impact of latency on throughput might be if we put request_to_send_more on the data socket.